src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -184,17 +184,22 @@
 (defn build-whitelist-authorizer
   "Build a function that will authorize requests based on the supplied
   certificate whitelist (see `cn-whitelist->authorizer` for more
-  details). Rejected requests are logged."
+  details). Returns :authorized if the request is allowed, otherwise a
+  string describing the reason not."
   [whitelist]
   {:pre  [(string? whitelist)]
    :post [(fn? %)]}
   (let [allowed? (kitchensink/cn-whitelist->authorizer whitelist)]
     (fn [{:keys [ssl-client-cn] :as req}]
       (if (allowed? req)
-        true
+        :authorized
         (do
           (log/warnf "%s rejected by certificate whitelist %s" ssl-client-cn whitelist)
-          false)))))
+          (format (str "The client certificate name (%s) doesn't "
+                       "appear on the certificate whitelist. Is your "
+                       "master's certname listed in PuppetDB's "
```

Comment:
This won't always be a master, in PE we have multiple services that connect to pdb.